### DISCUSSION TITLE
Subpages keep showing 404 when reload

### DISCUSSION THREAD

**Discussion by tokensWeb** [vercel/next.js#41097] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:35 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T8101
    Binaries:
      Node: 16.15.1
      npm: 8.11.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 12.3.1
      eslint-config-next: 12.3.1
      react: 18.1.0
      react-dom: 18.1.0

### What browser are you using? (if relevant)

Chrome 105.0.5195.125

### How are you deploying your application? (if relevant)

next build && next export

### Describe the Bug

subpages:
/test/[...id].tsx

showing 404 when reload, the page path: /test/1/1

pages
-- test
--- index.tsx
---[...id].tsx

### Expected Behavior

When refresh a sub page that it shows the sub page and not the 404 page.

Please note that when developing on localhost this all works fine. And this issue only occurs when a build is run and deployed to production server.

### Link to reproduction

https://codesandbox.io/s/festive-snyder-44954s?file=/pages/test/index.tsx:0-78

### To Reproduce

Create subfolders in the pages directory  e.g. test/[...id].tsx
Run next build && next export and deploy to server.
Go to a test/1/1 and reload the page and see if it loads the subpage or the 404 page.

**Comment by rubytree33** [comment]
In the docs for [static export](https://nextjs.org/docs/advanced-features/static-html-export) it says "Dynamic Routes when using `getStaticPaths`" are supported, but the reproduction doesn't have that. Maybe this is the problem?

**Comment by icyJoseph** [comment]
Hi,

In layman's terms, your production server "doesn't understand the paths".

The core of the "issue" is that when you `next export`, without `trailingSlashes`, then these files are generated:

```
➜ ls out/
404.html
_next
assets
blog.html
favicon.ico
index.html
static
vercel.svg
```

Notice that there's no `blog` page but `blog.html`, so when I navigate from `index.html` to `/blog` using the router, it is all good. However, when you reload there, you hit my server, and it finds no `blog` file or directory, so a `404.html` is served, depending on the server engine of course.

> It follows that you can load you page at `/blog.html`

There's two takes to this:

- Configure your server so that it can translate `/blog` to `/blog.html`, or at least look to see if that's available
   - You have to pretty much just Google around, or learn how your production server, say NGINX, works, and write a proper config file for it
- Use `trailingSlashes`, in `next.config.js`:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  trailingSlash: true,
  reactStrictMode: true,
};

module.exports = nextConfig;
```

Which when exported yields:

```
404
404.html
_next
assets
blog
favicon.ico
index.html
static
vercel.svg
```

And now blog is a folder, and inside it there's a file called `index.html`, so now reloading on `/blog` loads up that directory and it all works fine. However, **we must adjust our client side** links to use `/` at the end.
